push-sender/src/main/java/org/jboss/aerogear/unifiedpush/message/sender/apns/ApnsUtil.java

```diff
@@ -66,4 +73,32 @@ public static String readDefaultTopic(final byte[] keystore, final char[] passwo
 
         return null; // if no topic was found we try with null
     }
+
+    public static boolean checkValidity(final byte[] keystore, final char[] password) {
+        try {
+            final KeyStore keyStore = KeyStore.getInstance(KEYSTORE_TYPE);
+            keyStore.load(new ByteArrayInputStream(keystore), password);
+
+            final Enumeration<String> aliases = keyStore.aliases();
+
+            while (aliases.hasMoreElements()) {
+                final String alias = aliases.nextElement();
+                final X509Certificate certificate = (X509Certificate) keyStore.getCertificate(alias);
+
+
+                try{
+                    certificate.checkValidity();
```

Comment:
A .p12 can contain many certificates, even certificates we don't care at all. Do we want to fail if _any_ certificate inside the p12 is expired?

Reply:
@ziccardi Hmm, i would assume so since all of them should be issued by Apple for accessing APNS right? Or do you think there could be unrelated certs in there as well?

Reply:
@pb82 A P12 could contain any kind of certificate. I think we should at least check if the CN is related to sites we are interested in.